scripts/import-users.py

```diff
@@ -0,0 +1,410 @@
+
+    @staticmethod
+    def _parse_filename(filename):
+        """
+        Function to parse the filename to the project name.
+
+        Args:
+            filename:
+                The string of the filename of the TSV.
+        Returns:
+            string:
+                Returns a string that is AWS bucket naming compliant.
+        """
+        # First, break off file extension
+        filename = filename.replace(".tsv.gz", "")
+        filename = filename.replace(".tsv", "")
+        # Then create an intermediate with only valid characters
+        name = "".join(
+            [c for c in filename.replace("-", "") if c.isalnum() or c == "_"]
+        )
+        # Then clean up starting spaces
+        name = re.sub(r"^_*", "", name)
+        name = re.sub(r"_+", "-", name)
+        return name.lower()
+
+    @staticmethod
+    def _load_tsv(path):
+        """
+        Method to load in the pandas TSV, even if the TSV is just headers or not even headers.
+        Args:
+            path:
+                Absolute path to the (gzipped) TSV file.
+        Returns:
+            dataframe:
+                Returns a pandas.DataFrame containing the columns of a loaded in pandas TSV or
+                empty dataframe containing the correct columns.
+        """
+        try:
+            data = pd.read_csv(
+                path,
+                sep='\t',
+                dtype={
+                    "active": bool,
+                    "email": str,
+                    "firstName": str,
+                    "fullName": str,
+                    "lastName": str,
+                    "roles": str
+                }
+            ).fillna("")
+        except pd.errors.EmptyDataError:
+            data = pd.DataFrame(
+                columns=["active", "email", "firstName", "fullName", "lastName", "roles"]
+            )
+        return data
+
+
+class ProjectProcessor:
+    def __init__(self, api):
+        self.api = api
+        self.su = []
+        self.current_su = self._obtain_current_su(api)
+
+    def parse(self, cohort_data):
+        """
+        Bread and butter of the script. This function parses the cohort data dictionary into the
+        API, creating new projects and users alike. Also sets users admin rights accordingly,
+        as they were according to the export TSVs.
+
+        Args:
+            cohort_data:
+                Dictionary containing (key) the cohort name and (value) its user data as a pandas
+                dataframe.
+        """
+        for cohort_name, cohort_users in cohort_data.items():
+            # Skip users that are not categorized into a cohort
+            if cohort_name == "uncategorized-users":
+                continue
+            self._obtain_su(cohort_users)
+            self._put_project(name=cohort_name, users=cohort_users['email'].values.tolist())
+        self._process_uncategorized_users(cohort_data['uncategorized-users'])
+        self._add_admin_rights()
+
+    @staticmethod
```

Comment:
Why static ... it is called non-static no? Are the other @static* similar needed?

Reply:
As discussed: staticmethods in python are methods that do not really use anything of the class they are defined in, also pycharm complains about methods that do not use any "self" in their use (init variables, other methods etc.)